lib/utils/normalizeOptions.js

```diff
@@ -127,6 +128,48 @@ function normalizeOptions(compiler, options) {
   if (typeof options.compress === 'undefined') {
     options.compress = true;
   }
+
+  // if the user enables http2, we can safely enable https
+  if ((options.http2 && !options.https) || options.https === true) {
+    options.https = {
+      requestCert: false,
+    };
+  }
+
+  if (options.https) {
+    const getCertificate = require('./getCertificate');
+
+    for (const property of ['cacert', 'pfx', 'key', 'cert']) {
+      const value = options.https[property];
+      const isBuffer = value instanceof Buffer;
+
+      if (value && !isBuffer) {
+        let stats = null;
+
+        try {
+          stats = fs.lstatSync(fs.realpathSync(value)).isFile();
+        } catch (error) {
+          // ignore error
+        }
+
+        // It is file
+        options.https[property] = stats
+          ? fs.readFileSync(path.resolve(value))
+          : value;
+      }
+    }
+
+    let fakeCert;
+
+    if (!options.https.key || !options.https.cert) {
+      fakeCert = getCertificate(
+        compiler.getInfrastructureLogger('webpack-dev-server')
```

Comment:
We need assign to `this.logger` before and use `this.logger` everywhere

Reply:
fixed